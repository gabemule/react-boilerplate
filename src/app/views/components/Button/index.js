import React from 'react'

import { DsButton, DsParagraph, DsHeading } from '@gabemule/design-system'

const Button = () => {
  return (
    <div className={'Button'}>
      <DsHeading type="display">Buttons - Dynamic Theme Change</DsHeading>

      <br />
      <br />

      <DsParagraph type="small">Small</DsParagraph>

      <br />

      <DsButton size="small"> Primary </DsButton>
      <DsButton size="small" type="secondary">
        {' '}
        Secondary{' '}
      </DsButton>
      <DsButton size="small" type="warning">
        {' '}
        Warning{' '}
      </DsButton>
      <DsButton size="small" type="danger">
        {' '}
        Danger{' '}
      </DsButton>
      <DsButton disabled size="small">
        {' '}
        Disabled{' '}
      </DsButton>

      <br />
      <br />

      <DsParagraph type="small">Medium</DsParagraph>

      <br />

      <DsButton> Primary </DsButton>
      <DsButton type="secondary"> Secondary </DsButton>
      <DsButton type="warning"> Warning </DsButton>
      <DsButton type="danger"> Danger </DsButton>
      <DsButton disabled> Disabled </DsButton>

      <br />
      <br />

      <DsParagraph type="small">Large</DsParagraph>

      <br />

      <DsButton size="large"> Primary </DsButton>
      <DsButton size="large" type="secondary">
        {' '}
        Secondary{' '}
      </DsButton>
      <DsButton size="large" type="warning">
        {' '}
        Warning{' '}
      </DsButton>
      <DsButton size="large" type="danger">
        {' '}
        Danger{' '}
      </DsButton>
      <DsButton disabled size="large">
        {' '}
        Disabled{' '}
      </DsButton>
    </div>
  )
}

export default Button
