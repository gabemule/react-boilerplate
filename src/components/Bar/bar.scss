.bar {
  $self: &;

  background-color: #f2f2f2;
  #{ $self }__child-element {
    background: black;
  }
}
