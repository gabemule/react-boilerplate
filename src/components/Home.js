import React from "react";
import MyImage from "../assets/dog.jpg";

const Home = () => {
    return (
        <div>
            <h1>Home</h1>
            <img src={MyImage} alt={'home image'} />
        </div>
    );
};

export default Home;
