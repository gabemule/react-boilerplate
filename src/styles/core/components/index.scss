@import "Accordion/Accordion";
@import "Accordion/AccordionItem";
@import "Button/Button";
@import "Caption/Caption";
@import "GridLayout/GridLayout";
@import "Heading/Heading";
@import "Icon/Icon";
@import "Icon/Icons";
@import "Input/Input";
@import "Paragraph/Paragraph";
@import "Subtitle/Subtitle";
@import "ThemeProvider/ThemeProvider";
@import "Tooltip/Tooltip";