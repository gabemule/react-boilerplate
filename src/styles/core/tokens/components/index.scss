// components
@import "button";
@import "input";