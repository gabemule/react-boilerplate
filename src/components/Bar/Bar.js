import React from "react";

import "./bar.scss";
import PropTypes from "prop-types";

const Bar = ({children}) => {
  return (
    <div className={"bar"}>
      {children}
    </div>
  );
};

Bar.propTypes = {
  children: PropTypes.any,
};

export default Bar;
