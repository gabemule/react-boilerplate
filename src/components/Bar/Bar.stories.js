import React from 'react'
import { Bar } from '@/components/Bar'

const componentProps = {
  title: 'Components/Bar',
  component: Bar
}

export default componentProps

const Template = (args) => <Bar {...args}> Bar Component </Bar>

export const Default = Template.bind({})
