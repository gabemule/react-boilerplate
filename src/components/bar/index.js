import Bar from './bar';

export default Bar;
