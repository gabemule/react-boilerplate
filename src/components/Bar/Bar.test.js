import React from 'react'
import { shallow } from 'enzyme'

// Components
import { Bar } from '@/components/Bar'

// Constants
const BAR_COMPONENT_TEXT = 'Bar Component'
const BAR_COMPONENT_CLASS = '.bar'

/**
 * Setup method to avoid code repeat
 * @param props
 * @returns {{wrapper, props}}
 */
function setup(props) {
	const wrapper = shallow(<Bar>{BAR_COMPONENT_TEXT}</Bar>)
	return { wrapper, props }
}

describe('Bar Test Suite', () => {
	it('Should match snapshot', () => {
		expect(Bar).toMatchSnapshot()
	})
	it('Should have class', () => {
		const { wrapper } = setup()
		expect(wrapper.find(BAR_COMPONENT_CLASS).exists()).toBe(true)
	})
	it('Should have text', () => {
		const { wrapper } = setup()
		expect(wrapper.find(BAR_COMPONENT_CLASS).text()).toBe(BAR_COMPONENT_TEXT)
	})
})
