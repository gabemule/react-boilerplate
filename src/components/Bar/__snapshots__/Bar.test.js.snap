// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar Test Suite Should match snapshot 1`] = `[Function]`;
